client/cli/src/commands/inspect_node_key.rs

```diff
@@ -17,39 +17,67 @@
 
 //! Implementation of the `inspect-node-key` subcommand
 
-use crate::{Error, NetworkSchemeFlag};
+use crate::Error;
 use clap::Parser;
 use libp2p::identity::{ed25519, PublicKey};
-use std::{fs, path::PathBuf};
+use std::{
+	fs,
+	io::{self, Read},
+	path::PathBuf,
+};
 
 /// The `inspect-node-key` command
 #[derive(Debug, Parser)]
 #[clap(
 	name = "inspect-node-key",
-	about = "Print the peer ID corresponding to the node key in the given file."
+	about = "Load a node key from a file or stdin and print the corresponding peer-id."
 )]
 pub struct InspectNodeKeyCmd {
 	/// Name of file to read the secret key from.
+	///
+	/// If not given, the secret key is read from stdin (up to EOF).
 	#[clap(long)]
-	file: PathBuf,
+	file: Option<PathBuf>,
 
-	#[allow(missing_docs)]
-	#[clap(flatten)]
-	pub network_scheme: NetworkSchemeFlag,
```

Comment:
Are you sure no one uses it? How about adding a deprecation notice and then gracefully get rid of it?